File: matlab/appendix_2_3_problem2_depth_100_to_140.m
function [d,I,peak,peak_d]=appendix_2_3_problem2_depth_100_to_140
% 附录2.3：第二问，d = 100:1:140 的分段积分计算。
[d,I]=x4;
[peak,i]=max(I);
peak_d=d(i);

File: python/appendix_2_3_problem2_depth_100_to_140.py
from __future__ import annotations

from x4 import x4


def run():
    """Appendix 2.3: problem 2, d = 100:1:140."""
    return x4()


if __name__ == "__main__":
    d, I, peak, peak_d = run()
    print("d =", d)
    print("I =", I)
    print("peak =", peak)
    print("peak_d =", peak_d)
